### ISSUE TITLE
Freezes and slow performance

### ISSUE THREAD

**Issue by zeyangl** [Microsoft/vscode#9197]
- VSCode Version: 1.3.1
- OS Version: macOS 10.11.5
- Env:  working with Unity 5.3

Steps to Reproduce:

Frequent freezes working with Unity project(C#).
Particularly prominent when moving large chunk of code around. Background electron helper process runs at 100% cpu. It used to be much better before the VSCode 1.3 update.

It's hard to describe the exact steps since it may depend on my project size. But it should be pretty obvious in mid sized Unity project.

If this is the Intellisense running, it's probably good to have a bar indicator or something to make it more bearable.

**Comment by weinand** [comment]
@zeyangl what unity extensions are you using?

**Comment by zeyangl** [comment]
@weinand  If you mean how I open VSCode from Unity, I'm using dotBunny VSCode plugin by Matthew Davey. In VSCode I have OmniSharp C# and Unity Debugger plugin installed. Nothing else really.

**Comment by zeyangl** [comment]
I can double confirm that moving large chunk of code causes severe freezes, up to about 30 secs. It just happened to me again, multiple times, when I'm refactoring a large class to two files. I'm on a Mac Pro with 6 cores and 20G ram.

**Comment by weinand** [comment]
@zeyangl C# refactoring is done by the OmniSharp C# extension. I suggest to file an issue there: https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp

**Comment by zeyangl** [comment]
Just want to add some additional information.
I restarted VSCode and stuff appears to work properly for now. But after I opened the "Problems" window, the freezes come back again. Even after I close the "Problems" window, freezes remain.

**Comment by weinand** [comment]
@sandy081 any idea what this could be?

**Comment by sandy081** [comment]
@zeyangl Can you please try following

- Close the Problems panel and start moving the large chunk of code. Is there a freeze happening then? 
- Once move is finished, please open problems panel and see how many errors/warnings are generated. Note you can also get errors and warnings count in the status bar.

Thanks

**Comment by juol** [comment]
VSC 1.3.1
OS X 10.10.3

Very slow scrolling and general usability.

All extensions uninstalled.

Love VSC and this slow down seems to be pretty rare.

Keep up the great work guys!

Thanks.

**Comment by sandy081** [comment]
@juol Can you please let me know where is the slowness you are looking at? In Problems panel or other places? It would be great if you can provide us 
- Where are you seeing the slowness
- Are there any steps or scenarios those are causing this slowness

Thanks

**Comment by juol** [comment]
The slow down is most noticeable in the editor window. Scrolling and moving the caret as well as any selection operations.

When the 1.3.1 update came through the other day the update bar at the top of the editor window stuttered and scrolled down in a few jerky steps.

Expanding a file tree of moderate size is also slow.

It has this behaviour regardless of what directory or project I open.

Thanks

**Comment by sandy081** [comment]
@juol Extracted your comment as a separate issue - #9517. 

@zeyangl closing this assuming that above attached fix helps. Please try the latest insiders and let me know if it fixes your slowness issue.

Thanks.
